libs/designsystem/src/lib/components/item-sliding/item-sliding.component.ts

```diff
@@ -0,0 +1,40 @@
+import { Component, Input } from '@angular/core';
+
+import { EvaluatedListSwipeAction, ListSwipeAction } from '../list/list-swipe-action.type';
+
+export type ItemSlidingSide = 'left' | 'right';
+
+@Component({
+  selector: 'kirby-item-sliding',
+  templateUrl: './item-sliding.component.html',
+  styleUrls: ['./item-sliding.component.scss'],
+})
+export class ItemSlidingComponent {
+  _evaluatedSwipeActions: EvaluatedListSwipeAction[] = [];
+
+  @Input() set swipeActions(values: ListSwipeAction[]) {
+    this._evaluatedSwipeActions = values.map((value) => this.evaluateSwipeAction(value));
+  }
+
+  _side: 'start' | 'end' = 'start';
+  @Input() set side(value: ItemSlidingSide) {
+    this._side = value === 'left' ? 'start' : 'end';
+  }
+
+  private evaluateSwipeAction(swipeAction: ListSwipeAction): EvaluatedListSwipeAction {
+    /* TODO: is it possible to autogenerate these based on the type? */
+    const keysToExclude: (keyof ListSwipeAction)[] = ['onSelected', 'position'];
+
+    const evaluatedEntries = Object.entries(swipeAction).map(([key, value]) => ({
+      [key]: value instanceof Function && !keysToExclude.includes(key as any) ? value() : value,
+    }));
+
+    return Object.assign({}, ...evaluatedEntries);
+  }
```

Comment:
I wonder if we are jumping through too many hoops, and creating a bit much complexity for ourselves here?

The way I see it, this is to keep the external API for `ListSwipeAction` intact, which I really love. But, would an alternative be to simply change the `EvaluatedListSwipeAction` to something like `SwipeAction` and accept that teams would have to modify their code to support this way of using the sliding item?

The reason I got to think about this is the fact that I think the 'ListSwipeAction` is not a great name for something that is actually very item-centric. So maybe we could accept maintaining these two types, one for each of the ways to build out a list. 

I am truly in doubt.

Reply:
But that would mean that we will have a much harder time using this new sliding item in the old list, right?

Reply:
Warning! Stream of consciousness type'a'comment. 

No matter the conclusion, I will definitely rename `EvaluatedListSwipeaction` to `ItemSwipeAction` to make it more focused on item. 

I have until now (perhaps a bit blindly) just implemented the api from the original `ListSwipeAction`. Thinking a bit about it, I think we should remove the ability of passing in functions for the various values as we can no longer pass in an `item` for these functions. Which greatly reduces their usefulness and I can't actually see why you would use them. 

In the end, what the ability to pass functions give you, is to modify behaviour based on item. 

So the following: 

```typescript
swipeActions = {
   ...
   title: (item) => item.id % 2 === 0 ? `even: ${item.title}` : `odd: ${item.title}`
   ...
}; 
``` 

Would be equivalent to doing something like this (assuming swipeActions are ordered and id's are going from 0,1,2...swipeActions.length): 

HTML: 
```html
  <kirby-item-sliding
    *ngFor="let item of items; let idx = index"
    [swipeActions]="getSwipeActions(idx)"
  >
    <kirby-item>{{ item.value }}</kirby-item>
  </kirby-item-sliding>
```

TS: 
```typescript 
class SwipeExample {
    ...
    swipeActions = [{..., title: 'Hello'}, {..., title: 'World'}, {..., title: 'how'}, {..., title: 'ya'}, {..., title: 'doing?'}]

   getSwipeActions(idx: number): ListSwipeAction {
      const { title } = swipeActions[idx]
      return {
      ...
      title: idx % 2 === 0 ? `even: ${title}` : `odd: ${title}`
      ...
   }
}
```

As can be seen it is more work for the consumer. But that is part of the tradeoff when opening up and allowing consumers to construct items declaratively, instead of relying on them supplying item as a structured object that we then handle. 

... Thinking a bit more about it a solution that allows for both approaches, should be to allow for providing data for the callback via an input property, perhaps `contextData`? We would then just pass that data back when calling the functions in the ListSwipeActions. To recreate the above examples you would then: 

Template
```html
  <kirby-item-sliding
    *ngFor="let item of items;"
    [swipeActions]="swipeActions"
    [contextData]="item"
  >
    <kirby-item>{{ item.value }}</kirby-item>
  </kirby-item-sliding>
```

ItemSlidingComponent: 
```Typescript
 @Input() contextData: any; 

 private evaluateSwipeAction(swipeAction: ListSwipeAction): EvaluatedListSwipeAction {
    const keysToExclude: (keyof ListSwipeAction)[] = ['onSelected', 'position'];

     const evaluatedEntries = Object.entries(swipeAction).map(([key, value]) => ({
      // Note: the important change is on the below line: 
      [key]: value instanceof Function && !keysToExclude.includes(key as any) ? value(this.contextData) : value,
    }));

     return Object.assign({}, ...evaluatedEntries);
  }
```

Example Component (again, assuming swipeActions are ordered and id's are going from 0,1,2...swipeActions.length):  
```typescript
swipeActions = {
   ...
   title: (contextData) => contextData.id % 2 === 0 ? `even: ${contextData.title}` : `odd: ${contextData.title}`
   ...
}; 
```

Reply:
Having slept on this, I do not think we should create a contextData field until it has proven necessary. 

We can always add the option of having contextData - it is harder to remove it.